boards/PDM/Src/Io/Io_Efuse.c

```diff
@@ -0,0 +1,355 @@
+#include <assert.h>
+#include <math.h>
+#include <stdlib.h>
+#include "Io_Efuse.h"
+#include "configs/Io_EfuseConfig.h"
+
+#pragma GCC diagnostic ignored "-Wconversion"
+
+struct Efuse_Context
+{
+    SPI_HandleTypeDef *hspi;
+    GPIO_TypeDef *     nss_port;
+    uint16_t           nss_pin;
+
+    GPIO_TypeDef *fsob_port;
+    uint16_t      fsob_pin;
+    GPIO_TypeDef *fsb_port;
+    uint16_t      fsb_pin;
+
+    GPIO_TypeDef *channel0_port;
+    uint16_t      channel0_pin;
+    GPIO_TypeDef *channel1_port;
+    uint16_t      channel1_pin;
+
+    // The current state of the watchdog-in bit (bit 15). If the watchdog is
+    // enabled its state must be alternated at least once within the watchdog
+    // timeout period.
+    bool wdin_bit_to_set;
+};
+
+/**
+ * Calculate the parity of the SPI command to be sent to the Efuse and
+ * set/clear the parity bit. Using the XOR sum of bits method taken from:
+ * https://en.wikipedia.org/wiki/Parity_bit#Parity
```

Comment:
I'm assuming this is some form of error detection. Can we include that in the doc so we know what the parity bit is used for?

Reply:
💰

Reply:
Ya I believe parity bit is always used for error detection(https://en.wikipedia.org/wiki/Parity_bit), but I'll add a small comment

Reply:
^ oh I already have a link to the wikipedia page on parity bits